### ISSUE TITLE
Using spline_conv to approximate gradients on a graph

### ISSUE THREAD

**Issue by saitta-s** [rusty1s/pytorch_spline_conv#19]
Hi,
I was wondering if it is possible to manually calibrate the BSpline weights in order to replicate a Sobel-like kernel. My goal would be to have a kernel that when applied to every node feature returns the feature gradient (along a specified direction) at the node approximated as a convolution with the same feature on neighbors.
 If I understand correctly, it should in principle be possible to compute gradients on a graph in this way, but I think one needs to sort every node's neighbors in a consistent manner. 
Do you think this can be achieved with a spline_conv, or should I try to implement it from scratch? 

Many thanks.

**Comment by rusty1s** [comment]
I'm not sure why you think one needs sort a node's neighborhood. If I understand you correctly, you want to implement a `SplineConv` layer where B-spline weights models a Sobel-like kernel. You can do this by customizing the `SplineConv.weight`, which is a tensor of shape `[num_spline_weights, in_channels, out_channels]`. Therefore, for `in_channels = out_channels = 1`, that would be something like
```python
conv.weight = torch.tensor([-1, 0, 1, -2, 0, 2, -1, 0, 1]).view(-1, 1, 1)
```

**Comment by saitta-s** [comment]
Thanks for your prompt response. I tried what you suggested, but without the intended results.
My goal is to have a `SplineConv` layer that is given a feature on a graph and returns the feature's derivative along a specific spatial direction. 
In a sense, this would be analogous to a finite difference scheme, but generalized to unstructured meshes. This is why I thought that one needs to sort each node's neighborhood ([related paper](https://diglib.eg.org/bitstream/handle/10.2312/3dor20181060/111-114.pdf?sequence=1&isAllowed=y)).

Thanks again for the support.

**Comment by saitta-s** [comment]
I managed to do what I wanted using RBFs. I eventually implemented a RBF-FD network from scratch, but your repo proved to be useful in the process.
Thanks for the support anyway.
